### PR TITLE
Download cache

### PR DESCRIPTION
**Pull request by mshabunin**
Addition to opencv/opencv#8421

### CODE REVIEW

File: modules/xfeatures2d/cmake/download_boostdesc.cmake [hunk 1]
```diff
@@ -1,30 +1,34 @@
...
+  set(${status} TRUE PARENT_SCOPE)
+  foreach(id ${ids})
+    ocv_download(FILENAME ${name_${id}}
+                 HASH ${hash_${id}}
+                 URL ${BOOSTDESC_DOWNLOAD_URL}
+                 DESTINATION_DIR ${dst}
+                 RELATIVE_URL
+                 STATUS res)
+    if(NOT res)
+      set(${status} FALSE PARENT_SCOPE)
```
Comment (alalek) [inline modules/xfeatures2d/cmake/download_boostdesc.cmake]: break()

File: modules/dnn/cmake/download_protobuf.cmake [hunk 2]
```diff
@@ -1,51 +0,0 @@
-set(PROTOBUF_CPP_NAME "libprotobuf")
-set(PROTOBUF_CPP_DOWNLOAD_HASH "bd5e3eed635a8d32e2b99658633815ef")
-set(PROTOBUF_CPP_PATH "${CMAKE_CURRENT_BINARY_DIR}/3rdparty/protobuf/sources") # /protobuf-3.1.0 subdirectory
-
-set(OPENCV_PROTOBUF_CPP_DOWNLOAD_URL ${OPENCV_PROTOBUF_URL};$ENV{OPENCV_PROTOBUF_URL};https://github.com/google/protobuf/releases/download/)
```
Comment (alalek) [inline modules/dnn/cmake/download_protobuf.cmake]: URL override

File: modules/xfeatures2d/cmake/download_boostdesc.cmake [hunk 3]
```diff
@@ -1,30 +1,34 @@
...
+  set(hash_BGM_HD "324426a24fa56ad9c5b8e3e0b3e5303e")
+  set(hash_BINBOOST_064 "202e1b3e9fec871b04da31f7f016679f")
+  set(hash_BINBOOST_128 "98ea99d399965c03d555cef3ea502a0b")
+  set(hash_BINBOOST_256 "e6dcfa9f647779eb1ce446a8d759b6ea")
+  set(hash_LBGM "0ae0675534aa318d9668f2a179c2a052")
 
+  set(BOOSTDESC_DOWNLOAD_URL "https://raw.githubusercontent.com/opencv/opencv_3rdparty/${OPENCV_3RDPARTY_COMMIT}/")
 
-
-set(BOOSTDESC_DOWNLOAD_URL ${OPENCV_CONTRIB_BOOSTDESC_URL};$ENV{OPENCV_CONTRIB_BOOSTDESC_URL};https://raw.githubusercontent.com/opencv/opencv_3rdparty/${OPENCV_3RDPARTY_COMMIT}/)
```
Comment (alalek) [inline modules/xfeatures2d/cmake/download_boostdesc.cmake]: URL override

File: modules/xfeatures2d/cmake/download_vgg.cmake [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
+  set(name_VGG_80 "vgg_generated_80.i")
+  set(name_VGG_120 "vgg_generated_120.i")
+  set(hash_VGG_48 "e8d0dcd54d1bcfdc29203d011a797179")
+  set(hash_VGG_64 "7126a5d9a8884ebca5aea5d63d677225")
+  set(hash_VGG_80 "7cd47228edec52b6d82f46511af325c5")
+  set(hash_VGG_120 "151805e03568c9f490a5e3a872777b75")
 
+  set(VGG_DOWNLOAD_URL "https://raw.githubusercontent.com/opencv/opencv_3rdparty/${OPENCV_3RDPARTY_COMMIT}/")
 
-set(VGG_DOWNLOAD_URL ${OPENCV_CONTRIB_VGG_URL};$ENV{OPENCV_CONTRIB_VGG_URL};https://raw.githubusercontent.com/opencv/opencv_3rdparty/${OPENCV_3RDPARTY_COMMIT}/)
```
Comment (alalek) [inline modules/xfeatures2d/cmake/download_vgg.cmake]: URL override

File: modules/xfeatures2d/cmake/download_boostdesc.cmake (hunk 1 above)
Comment (mshabunin) [inline modules/xfeatures2d/cmake/download_boostdesc.cmake]: This loop is continued intentionally to generate complete download log file.

**Comment by alalek** [comment]
:+1:
Please squash commits into one.

**Pull request merged**